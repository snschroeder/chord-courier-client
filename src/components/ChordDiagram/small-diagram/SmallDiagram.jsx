/* eslint-disable react/no-array-index-key */
/* eslint-disable array-callback-return */
/* eslint-disable no-undef */
import React from 'react';
import PropTypes from 'prop-types';
import SmallMutedString from './small-muted-string/SmallMutedString';
import SmallOpenString from './small-open-string/SmallOpenString';
import LittleFinger from './little-finger/LittleFinger';
import SmallFret from './small-fret/SmallFret';
import './small-diagram.css';

export default function SmallDiagram(props) {
  const generateFingerPosition = (position) => {
    let fingerPos = position.split(',');
    const start = fingerPos[0];
    const startFret = start !== '0' ? 'small-diagram' : 'small-open-diagram';
    let barre = '';
    fingerPos = fingerPos.map((pos) => (!'xbo'.includes(pos) ? parseInt(pos, 10) : pos));

    // eslint-disable-next-line consistent-return
    fingerPos = fingerPos.map((pos, index) => {
      if (pos === 'b') {
        barre = 'small-barre';
      } if (pos === 'x') {
        return <SmallMutedString stringNo={index} key={index} />;
      } if (pos === 'o') {
        return <SmallOpenString stringNo={index} key={index} />;
      } if (index === 0 && pos !== 0) {
        return <SmallFret fret={pos} key={index} />;
      } if (index >= 1 && typeof pos === 'number') {
        return <LittleFinger stringNo={index} fret={pos} key={index} />;
      }
    });
    return { startFret, fingerPos, barre };
  };

  const { chordCode } = props;
  const diagramDetails = generateFingerPosition(chordCode);

  return (
    <div className={diagramDetails.startFret}>
      <div className={diagramDetails.barre} />
      {diagramDetails.fingerPos}
    </div>
  );
}

SmallDiagram.defaultProps = {
  chordCode: '0,x,2,1,o,0,o',
};

SmallDiagram.propTypes = {
  chordCode: PropTypes.string,
};
