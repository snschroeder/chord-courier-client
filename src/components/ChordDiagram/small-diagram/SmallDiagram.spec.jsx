/* eslint-disable no-undef */
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import SmallDiagram from './SmallDiagram';

describe('SmallDiagram component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <SmallDiagram chordCode="0,x,2,1,o,0,o" />
      </BrowserRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders open position C maj chord when fed correct props', () => {
    const wrapper = shallow(<SmallDiagram chordCode="0,x,2,1,o,0,o" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
