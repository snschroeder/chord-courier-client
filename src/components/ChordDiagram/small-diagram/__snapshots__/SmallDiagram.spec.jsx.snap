// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallDiagram component renders open position C maj chord when fed correct props 1`] = `
<div
  className="small-open-diagram"
>
  <div
    className=""
  />
  <SmallMutedString
    key="1"
    stringNo={1}
  />
  <LittleFinger
    fret={2}
    key="2"
    stringNo={2}
  />
  <LittleFinger
    fret={1}
    key="3"
    stringNo={3}
  />
  <SmallOpenString
    key="4"
    stringNo={4}
  />
  <LittleFinger
    fret={0}
    key="5"
    stringNo={5}
  />
  <SmallOpenString
    key="6"
    stringNo={6}
  />
</div>
`;
